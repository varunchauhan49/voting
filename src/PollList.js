import React, { Component } from 'react';
import { Doughnut,defaults} from 'react-chartjs-2';
import { Row, Col } from 'reactstrap';

// defaults.labelFontSize = 24;
defaults.global.legend.display = false;

const doughnut = {
  datasets: [
    {
      data: [50,50],
      backgroundColor: [
        '#77aac6',
        '#d95e07',
      ],
      hoverBackgroundColor: [
        '#77aac6',
        '#d95e07',
      ],
    }],
};

export default function PollList () {
    let pollTitleDiv = {
      padding:'10px',
      textAlign:'left',
      color:'rgba(16, 58, 109, 0.85)'
    };
    let pollTitleCss = {
      marginLeft: '10px',
      fontSize:'22px',
      fontWeight:'600'
    };
    let pollHeadingDiv = {
      padding:'0px 20px 20px 30px',
      textAlign:'left',
    }
    let currDate = new Date(1516605447 * 1000);
    let monthsArr = ['Jan','Feb','Mar','Apr','May','Jun','Jul','Aug','Sep','Oct','Nov','Dec'];
    let dateString = currDate.getDate() + ' ' + monthsArr[currDate.getMonth()] + ' ' + currDate.getFullYear();
    return (
      <Row>
        <Col xs="12" sm="4">
          <Doughnut data={doughnut}  />
        </Col>
        <Col xs="12" sm="8">
          <div id="pollTitleDiv" key="pollTitleDiv" style={pollTitleDiv}>
            <i className="fa fa-comment" aria-hidden="true"></i>
            <span style={pollTitleCss}>{dateString}</span>
          </div>
          <div id="pollHeadingDiv" key="pollHeadingDiv" style={pollHeadingDiv}>
            <span>Is bitcoin worth the time and money that mining requires?</span>
          </div>
        </Col>
        <hr />
      </Row>

    );
  };
