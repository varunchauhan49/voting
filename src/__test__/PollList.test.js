import { shallow } from 'enzyme';
import PollList from '../PollList.js';
import React from 'react';

describe('PollList', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<PollList />);
    expect(wrapper).toMatchSnapshot();
  });
});