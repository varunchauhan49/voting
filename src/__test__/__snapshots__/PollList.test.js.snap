// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PollList renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PollList />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Col
          sm="4"
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
          xs="12"
        >
          <Doughnut
            data={
              Object {
                "datasets": Array [
                  Object {
                    "backgroundColor": Array [
                      "#77aac6",
                      "#d95e07",
                    ],
                    "data": Array [
                      50,
                      50,
                    ],
                    "hoverBackgroundColor": Array [
                      "#77aac6",
                      "#d95e07",
                    ],
                  },
                ],
              }
            }
          />
        </Col>,
        <Col
          sm="8"
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
          xs="12"
        >
          <div
            id="pollTitleDiv"
            style={
              Object {
                "color": "rgba(16, 58, 109, 0.85)",
                "padding": "10px",
                "textAlign": "left",
              }
            }
          >
            <i
              aria-hidden="true"
              className="fa fa-comment"
            />
            <span
              style={
                Object {
                  "fontSize": "22px",
                  "fontWeight": "600",
                  "marginLeft": "10px",
                }
              }
            >
              22 Jan 2018
            </span>
          </div>
          <div
            id="pollHeadingDiv"
            style={
              Object {
                "padding": "0px 20px 20px 30px",
                "textAlign": "left",
              }
            }
          >
            <span>
              Is bitcoin worth the time and money that mining requires?
            </span>
          </div>
        </Col>,
        <hr />,
      ],
      "tag": "div",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Doughnut
            data={
              Object {
                "datasets": Array [
                  Object {
                    "backgroundColor": Array [
                      "#77aac6",
                      "#d95e07",
                    ],
                    "data": Array [
                      50,
                      50,
                    ],
                    "hoverBackgroundColor": Array [
                      "#77aac6",
                      "#d95e07",
                    ],
                  },
                ],
              }
            }
          />,
          "sm": "4",
          "tag": "div",
          "widths": Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ],
          "xs": "12",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "data": Object {
              "datasets": Array [
                Object {
                  "backgroundColor": Array [
                    "#77aac6",
                    "#d95e07",
                  ],
                  "data": Array [
                    50,
                    50,
                  ],
                  "hoverBackgroundColor": Array [
                    "#77aac6",
                    "#d95e07",
                  ],
                },
              ],
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <div
              id="pollTitleDiv"
              style={
                Object {
                  "color": "rgba(16, 58, 109, 0.85)",
                  "padding": "10px",
                  "textAlign": "left",
                }
              }
            >
              <i
                aria-hidden="true"
                className="fa fa-comment"
              />
              <span
                style={
                  Object {
                    "fontSize": "22px",
                    "fontWeight": "600",
                    "marginLeft": "10px",
                  }
                }
              >
                22 Jan 2018
              </span>
            </div>,
            <div
              id="pollHeadingDiv"
              style={
                Object {
                  "padding": "0px 20px 20px 30px",
                  "textAlign": "left",
                }
              }
            >
              <span>
                Is bitcoin worth the time and money that mining requires?
              </span>
            </div>,
          ],
          "sm": "8",
          "tag": "div",
          "widths": Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ],
          "xs": "12",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "pollTitleDiv",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <i
                  aria-hidden="true"
                  className="fa fa-comment"
                />,
                <span
                  style={
                    Object {
                      "fontSize": "22px",
                      "fontWeight": "600",
                      "marginLeft": "10px",
                    }
                  }
                >
                  22 Jan 2018
                </span>,
              ],
              "id": "pollTitleDiv",
              "style": Object {
                "color": "rgba(16, 58, 109, 0.85)",
                "padding": "10px",
                "textAlign": "left",
              },
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "aria-hidden": "true",
                  "className": "fa fa-comment",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "22 Jan 2018",
                  "style": Object {
                    "fontSize": "22px",
                    "fontWeight": "600",
                    "marginLeft": "10px",
                  },
                },
                "ref": null,
                "rendered": "22 Jan 2018",
                "type": "span",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "pollHeadingDiv",
            "nodeType": "host",
            "props": Object {
              "children": <span>
                Is bitcoin worth the time and money that mining requires?
              </span>,
              "id": "pollHeadingDiv",
              "style": Object {
                "padding": "0px 20px 20px 30px",
                "textAlign": "left",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Is bitcoin worth the time and money that mining requires?",
              },
              "ref": null,
              "rendered": "Is bitcoin worth the time and money that mining requires?",
              "type": "span",
            },
            "type": "div",
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Col
            sm="4"
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
            xs="12"
          >
            <Doughnut
              data={
                Object {
                  "datasets": Array [
                    Object {
                      "backgroundColor": Array [
                        "#77aac6",
                        "#d95e07",
                      ],
                      "data": Array [
                        50,
                        50,
                      ],
                      "hoverBackgroundColor": Array [
                        "#77aac6",
                        "#d95e07",
                      ],
                    },
                  ],
                }
              }
            />
          </Col>,
          <Col
            sm="8"
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
            xs="12"
          >
            <div
              id="pollTitleDiv"
              style={
                Object {
                  "color": "rgba(16, 58, 109, 0.85)",
                  "padding": "10px",
                  "textAlign": "left",
                }
              }
            >
              <i
                aria-hidden="true"
                className="fa fa-comment"
              />
              <span
                style={
                  Object {
                    "fontSize": "22px",
                    "fontWeight": "600",
                    "marginLeft": "10px",
                  }
                }
              >
                22 Jan 2018
              </span>
            </div>
            <div
              id="pollHeadingDiv"
              style={
                Object {
                  "padding": "0px 20px 20px 30px",
                  "textAlign": "left",
                }
              }
            >
              <span>
                Is bitcoin worth the time and money that mining requires?
              </span>
            </div>
          </Col>,
          <hr />,
        ],
        "tag": "div",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Doughnut
              data={
                Object {
                  "datasets": Array [
                    Object {
                      "backgroundColor": Array [
                        "#77aac6",
                        "#d95e07",
                      ],
                      "data": Array [
                        50,
                        50,
                      ],
                      "hoverBackgroundColor": Array [
                        "#77aac6",
                        "#d95e07",
                      ],
                    },
                  ],
                }
              }
            />,
            "sm": "4",
            "tag": "div",
            "widths": Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ],
            "xs": "12",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "data": Object {
                "datasets": Array [
                  Object {
                    "backgroundColor": Array [
                      "#77aac6",
                      "#d95e07",
                    ],
                    "data": Array [
                      50,
                      50,
                    ],
                    "hoverBackgroundColor": Array [
                      "#77aac6",
                      "#d95e07",
                    ],
                  },
                ],
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <div
                id="pollTitleDiv"
                style={
                  Object {
                    "color": "rgba(16, 58, 109, 0.85)",
                    "padding": "10px",
                    "textAlign": "left",
                  }
                }
              >
                <i
                  aria-hidden="true"
                  className="fa fa-comment"
                />
                <span
                  style={
                    Object {
                      "fontSize": "22px",
                      "fontWeight": "600",
                      "marginLeft": "10px",
                    }
                  }
                >
                  22 Jan 2018
                </span>
              </div>,
              <div
                id="pollHeadingDiv"
                style={
                  Object {
                    "padding": "0px 20px 20px 30px",
                    "textAlign": "left",
                  }
                }
              >
                <span>
                  Is bitcoin worth the time and money that mining requires?
                </span>
              </div>,
            ],
            "sm": "8",
            "tag": "div",
            "widths": Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ],
            "xs": "12",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "pollTitleDiv",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <i
                    aria-hidden="true"
                    className="fa fa-comment"
                  />,
                  <span
                    style={
                      Object {
                        "fontSize": "22px",
                        "fontWeight": "600",
                        "marginLeft": "10px",
                      }
                    }
                  >
                    22 Jan 2018
                  </span>,
                ],
                "id": "pollTitleDiv",
                "style": Object {
                  "color": "rgba(16, 58, 109, 0.85)",
                  "padding": "10px",
                  "textAlign": "left",
                },
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "aria-hidden": "true",
                    "className": "fa fa-comment",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "22 Jan 2018",
                    "style": Object {
                      "fontSize": "22px",
                      "fontWeight": "600",
                      "marginLeft": "10px",
                    },
                  },
                  "ref": null,
                  "rendered": "22 Jan 2018",
                  "type": "span",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": "pollHeadingDiv",
              "nodeType": "host",
              "props": Object {
                "children": <span>
                  Is bitcoin worth the time and money that mining requires?
                </span>,
                "id": "pollHeadingDiv",
                "style": Object {
                  "padding": "0px 20px 20px 30px",
                  "textAlign": "left",
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Is bitcoin worth the time and money that mining requires?",
                },
                "ref": null,
                "rendered": "Is bitcoin worth the time and money that mining requires?",
                "type": "span",
              },
              "type": "div",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
